### ISSUE TITLE
meta-extract fails with minimal arguments

### ISSUE THREAD

**Issue by mih** [datalad/datalad-metalad#75]
I tried

```
% datalad meta-extract metalad_core_file
[ERROR  ] one of the hex, bytes, bytes_le, fields, or int arguments must be given [uuid.py:__init__:171] (TypeError) 
```

**Comment by christian-monch** [comment]
That is a bug. It is due to invalid verification of dataset-existence on a given path. Have switched to verifying "Dataset.id", not sure that is the best way though.

**Comment by mih** [comment]
The standard approach is `require_dataset()`, but it seems you need more than what it does.

**Comment by christian-monch** [comment]
I thought the same, but it seems that require_dataset is happy to return a dataset, if a valid git-repo is in the provided path

**Comment by mih** [comment]
In general, the goal is to be as functional as possible, with as little datalad as possible. It seems that the need for a datalad ID is for having an identifier for a dataset. It might be worth thinking about being able to derive an identifier for a plain Git repository too. Having a hard requirement for all repos to also be datalad datasets will automatically exclude all 3rd-party source code repositories that are included as a subdataset. I personally do that a lot, and I don't see why a datalad (UU)ID is better than, say, the hexsha of the most recent commit. I may not see the full picture yet though...

**Comment by mih** [comment]
I tried again today on current master dc48f104fb61df2b4b4ca4b815ac282eb0a6e261

The error is different, but still matches the title of the issue:

```
% dl --dbg meta-extract metalad_core_file
Traceback (most recent call last):
  File "/home/mih/env/datalad-dev/bin/datalad", line 33, in <module>
    sys.exit(load_entry_point('datalad', 'console_scripts', 'datalad')())
  File "/home/mih/hacking/datalad/git/datalad/cmdline/main.py", line 440, in main
    ret = cmdlineargs.func(cmdlineargs)
  File "/home/mih/hacking/datalad/git/datalad/interface/base.py", line 802, in call_from_parser
    ret = list(ret)
  File "/home/mih/hacking/datalad/git/datalad/interface/utils.py", line 389, in generator_func
    for r in _process_results(
  File "/home/mih/hacking/datalad/git/datalad/interface/utils.py", line 557, in _process_results
    for res in results:
  File "/home/mih/hacking/datalad/metalad/datalad_metalad/extract.py", line 235, in __call__
    yield from do_dataset_extraction(extraction_parameters)
  File "/home/mih/hacking/datalad/metalad/datalad_metalad/extract.py", line 281, in do_dataset_extraction
    assert issubclass(ep.extractor_class, DatasetMetadataExtractor)
AssertionError

> /home/mih/hacking/datalad/metalad/datalad_metalad/extract.py(281)do_dataset_extraction()
-> assert issubclass(ep.extractor_class, DatasetMetadataExtractor)
```

**Comment by christian-monch** [comment]
Fixed by PR #107

**Comment by christian-monch** [comment]
This issue if fixed by PR #107.

I created issue #108 to continue the ID discussion
